icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/CjMLBreakEngine.java

```diff
@@ -0,0 +1,346 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.impl.breakiter;
+
+import static com.ibm.icu.impl.CharacterIteration.current32;
+import static com.ibm.icu.impl.CharacterIteration.next32;
+
+import com.ibm.icu.impl.ICUData;
+import com.ibm.icu.lang.UCharacter;
+import com.ibm.icu.text.UnicodeSet;
+import com.ibm.icu.util.UResourceBundle;
+import com.ibm.icu.util.UResourceBundleIterator;
+
+import java.text.CharacterIterator;
+import java.util.ArrayList;
+import java.util.Hashtable;
+
+public class CjMLBreakEngine {
+
+    private static final int INVALID = 0x7c; // '|';
+    private UnicodeSet fDigitOrOpenPunctuationOrAlphabetSet;
+    private UnicodeSet fClosePunctuationSet;
+    private Hashtable<String, Integer> fModel;
```

Comment:
Any reason to use Hashtable over HashMap?

Reply:
Thanks. HashMap is enough in a single thread environment.
Change to use it.